pkg/timeline/service/account.ts

```diff
@@ -0,0 +1,66 @@
+import { Ether, Result } from '@mikuroxina/mini-fn';
+
+import type { AccountID } from '../../accounts/model/account.js';
+import type { ID } from '../../id/type.js';
+import type { Note } from '../../notes/model/note.js';
+import {
+  type FetchAccountTimelineFilter,
+  type TimelineRepository,
+  timelineRepoSymbol,
+} from '../model/repository.js';
+import {
+  type NoteVisibilityService,
+  noteVisibilitySymbol,
+} from './noteVisibility.js';
+
+export class AccountTimelineService {
+  private readonly noteVisibilityService: NoteVisibilityService;
+  private readonly timelineRepository: TimelineRepository;
+
+  constructor(args: {
+    noteVisibilityService: NoteVisibilityService;
+    timelineRepository: TimelineRepository;
+  }) {
+    this.noteVisibilityService = args.noteVisibilityService;
+    this.timelineRepository = args.timelineRepository;
+  }
+
+  async handle(
+    targetId: ID<AccountID>,
+    filter: FetchAccountTimelineFilter,
+  ): Promise<Result.Result<Error, Note[]>> {
+    const res = await this.timelineRepository.getAccountTimeline(
+      targetId,
+      filter,
+    );
+    if (Result.isErr(res)) {
+      return Result.err(res[1]);
+    }
+
+    // NOTE: AccountTimeline not include direct note
+    const directFiltered = res[1].filter((v) => v.getVisibility() !== 'DIRECT');
+
+    const filtered: Note[] = [];
+    for (const v of directFiltered) {
+      const visibility = await this.noteVisibilityService.handle({
+        accountID: filter.id,
+        note: v,
+      });
+      if (visibility) {
```

Comment:
```suggestion
      const isVisible = await this.noteVisibilityService.handle({
        accountID: filter.id,
        note: v,
      });
      if (isVisible) {
```
`visibility` だと `true` と `false` のうちどちらの値が見えることを表すのか分からないと思いました. `isVisible` のような変数名にしたり, `NoteVisibilityService.handle` メソッドに戻り値の意味等に関するドキュメントコメントを追加してみてはどうでしょうか.